python/dataproc_templates/hive/util/hive_ddl_extractor.py

```diff
@@ -0,0 +1,170 @@
+# limitations under the License.
+#
+
+
+import argparse
+import pprint
+import sys
+import io
+from logging import Logger
+from typing import Any, Dict, Optional, Sequence
+
+import dataproc_templates.util.template_constants as constants
+import pandas as pd
+from dataproc_templates import BaseTemplate
+from google.cloud import storage
+import pyspark
+from pyspark.sql import DataFrameWriter, SparkSession
+from google.cloud import bigquery
+from google.cloud.bigquery.table import Table
+from google.cloud.bigquery.dataset import Dataset
+from datetime import datetime
+
+
+def WriteToCloud (ddls_to_rdd, bucket, path):
+    # Write
+    now = datetime.now()
+    timedate = now.strftime("%m-%d-%Y %H.%M.%S")
+    ddls_to_rdd.coalesce(1).saveAsTextFile("gs://"+bucket+"/"+path+timedate)
+
+class HiveDDLExtractorTemplate(BaseTemplate): 
+    """
+    Dataproc template implementing exports from Hive to BigQuery
+    """
+
+    @staticmethod
+    def parse_args(args: Optional[Sequence[str]] = None) -> Dict[str, Any]:
+        parser: argparse.ArgumentParser = argparse.ArgumentParser()
+
+        parser.add_argument(
+            f'--{constants.HIVE_DDL_EXTRACTOR_INPUT_DATABASE}',
+            dest=constants.HIVE_DDL_EXTRACTOR_INPUT_DATABASE,
+            required=True,
+            help='Hive database for importing data to BigQuery'
+        )
+
+        parser.add_argument(
+            f'--{constants.HIVE_DDL_EXTRACTOR_OUTPUT_BUCKET}',
+            dest=constants.HIVE_DDL_EXTRACTOR_OUTPUT_BUCKET,
+            required=True,
+            help='Bucket for the output'
+        )
+
+        parser.add_argument(
+            f'--{constants.HIVE_DDL_EXTRACTOR_OUTPUT_DATASET}',
+            dest=constants.HIVE_DDL_EXTRACTOR_OUTPUT_DATASET,
+            required=True,
+            help='BigQuery output dataset name'
+        )
+        parser.add_argument(
+            f'--{constants.HIVE_DDL_EXTRACTOR_OUTPUT_TABLE}',
+            dest=constants.HIVE_DDL_EXTRACTOR_OUTPUT_TABLE,
+            required=True,
+            help='BigQuery output table name'
+        )
+       
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+        """
+        
+        Dataproc template allowing the extraction of DDLs from Hive Metastore
+
+        """
+        logger: Logger = self.get_logger(spark=spark)
+
+        hive_database: str = args[constants.HIVE_DDL_EXTRACTOR_INPUT_DATABASE]
+        gcs_working_directory: str = args[constants.HIVE_DDL_EXTRACTOR_OUTPUT_BUCKET]
+        bigquery_dataset: str = args[constants.HIVE_DDL_EXTRACTOR_OUTPUT_DATASET]
+        bigquery_table: str = args[constants.HIVE_DDL_EXTRACTOR_OUTPUT_TABLE]
+        gcs_target_path = constants.HIVE_DDL_EXTRACTOR_OUTPUT_PATH
+        gcs_staging_path="gs://"+gcs_working_directory+"/RawZone/"
```

Comment:
I suggest you expect the full bucket + path from the user. The user should input for example "gs://my_bucket/hiveddls/". That is because all other templates writing to GCS work like this, and here you are expecting only the bucket name. You should expect only the bucket name for the BigQuery temporary bucket. The user should provide 2 parameters: one for BigQuery temporary bucket, and one for the GCS output location.
For example, see GCS to BigQuery parameters:
...
--gcs.bigquery.input.location="gs://<my_bucket>/<path_input_files>/" \
...
--gcs.bigquery.temp.bucket.name="<maybe_other_bucket_name>"
...